libs/top_level_metrics.py

```diff
@@ -104,6 +105,25 @@ def calculate_metrics_for_timeseries(
     return metrics, metric_summary
 
 
+def calculate_or_copy_test_positivity(data):
+    # Use POSITIVE_TESTS and NEGATIVE_TEST if they are recent or TEST_POSITIVITY is not available
+    # for this region.
+    positive_negative_recent = has_data_in_past_10_days(
+        data[CommonFields.POSITIVE_TESTS]
+    ) and has_data_in_past_10_days(data[CommonFields.NEGATIVE_TESTS])
+    test_positivity = common_df.get_timeseries(data, CommonFields.TEST_POSITIVITY, EMPTY_TS)
+    if positive_negative_recent or not test_positivity.notna().any():
```

Comment:
FWIW- This logic is kinda' moot because if the data is older than 7 days it won't be used (on the website at least) anyway. https://github.com/covid-projections/covid-data-model/blob/094c13d3872830baec41b24c299a23d878d8c002/libs/top_level_metrics.py#L213

But I think this code needs to be reworked anyway. I think we're going to want to have a different set of calculation methods for states and for counties, since we know the HHS data lets us (almost) always get test specimens and so we should just unify on that...  and ideally the HHS data would just flow through your test positivity calculation instead of hitting a different codepath here.

But maybe we should just get this committed in its current form and iterate.

Reply:
Yes, for sure I'd like to extract the calculations from top_level_metrics so it gets only the value it needs to copy to the API.